.buildkite/validatePipeline.ts

```diff
@@ -26,6 +26,44 @@ pipelines.forEach((pipeline) => {
 
     errors.push({ pipeline, message });
   }
+
+  const keys: string[] = [];
+  const keyRefs: string[] = [];
+
+  // Check for duplicate and invalid keys
+  parsed.steps.forEach(
+    (step: string | { key?: string; depends_on?: string[] | string }) => {
```

Comment:
If we do more of this kinda thing, we might consider adding some schema generated types.